Created printer configuration for Voron Zero

FrostiFish opened the pull request:
I created a printer config to use Full Control with my Voron 0.1. It uses the print_start and print_end macros included in the klipper printer.cfg for the V0. It should work with all V0 versions. 
I did comment out the default priming blob as my print_start macro has a prime line included. I'm not sure what priming routine would be appropriate as the V0 has a very small bed. I'm open to ideas on that and happy to include it in my fork.
An optional z_offset setting was also added.

fullcontrol-xyz commented:
Excellent! Thanks very much. 

With regards to the default priming blob, I removed it from the ultimaker2plus for a similar reason. Did you personally change your print_start macro to have the prime line included? I.e. will pretty much all voron users have a prime line in their macro and therefore not need the priming blob?

FrostiFish commented:
The printer config has the macro defined as follows:
[gcode_macro PRINT_START]
#   Use PRINT_START for the slicer starting script - please customize for your slicer of choice
gcode:
    G28                            ; home all axes
    G90                            ; absolute positioning    
    G1 Z20 F3000                   ; move nozzle away from bed
(from: https://github.com/VoronDesign/Voron-0/tree/Voron0.2/Firmware)

Priming is not configured by default. I know some people use something like a nozzle brush and purge bucket for priming.
But I'd assume there's a reasonable amount of people who configure this in their slicer. I'll make a print override so the default full contol priming sequence can be disabled, if one desires. But make it enabled by default.

I'll also update the z offset to not override the set z offset if the user doesn't provide any. Currently it set the z offset to 0 if none is provided by the user.

FrostiFish commented:
@fullcontrol-xyz I updated the file to optionally disable the prime blob. I have yet to run it through some tests for validation, but it should work. Could you have a look at the overall file? I think the way I implemented overrides (for the default prime blob, and z-offset) might be useful for more printers than just the Voron Zero

AndyGlx commented:
Yep! I'm hoping to look at it tmrw. Thanks again for doing it!

AndyGlx commented:
I've merged it and made a couple of minor changes in a subsequent commit.

I changed the default printer override to be `'primer': 'travel'` instead of `'primer': 'no_primer'`

This is because if you set `'default_primer': False`, there are no Points in the start gcode and FullControl has no information about where the nozzle is before the first point. For a travel move, this doesn't matter, but for a printing move, it needs to know where the nozzle is so that the correct extrusion volume can be calculated for the line that prints to the first point in the design. So previously the nozzle moved to the first point without any extrusion, which could have led to drooling, but that is no longer the case. 

Also, I change `default_primer` to be `include_purge` to avoid confusion about there being two primers. In FullControl, 'primer' has so far referred to the printing of lines immediately before the main printed object. So, by calling the blobby thing a 'purge', it will hopefully be more clear to people.

If you think either of those changes isn't good, let me know and I can revert or revise them.

Thanks again!

FrostiFish commented:
I'm glad to have helped :)
I will share this in the Voron Discord to hopefully get more people enthused about Full Control!
The name changes make perfect sense, it's less confusing this way.
